Sources/Sentry/Profiling/SentryLaunchProfiling.m

```diff
@@ -135,28 +117,33 @@
 
         SENTRY_LOG_INFO(@"Starting app launch profile at %llu", appLaunchSystemTime);
 
-        // don't worry about synchronizing the write here, as there should be no other tracing
-        // activity going on this early in the process. this codepath is also behind a dispatch_once
-        isTracingAppLaunch = [SentryProfiler startWithTracer:appLaunchTraceId];
+        SentryTransactionContext *context =
+            [[SentryTransactionContext alloc] initWithName:@"launch" operation:@"app.lifecycle"];
+        SentryTracerConfiguration *config = [SentryTracerConfiguration defaultConfiguration];
+        NSDictionary<NSString *, NSNumber *> *rates = appLaunchProfileConfiguration();
+        NSNumber *profilesRate = rates[kSentryLaunchProfileConfigKeyProfilesSampleRate];
+        NSNumber *tracesRate = rates[kSentryLaunchProfileConfigKeyTracesSampleRate];
+        if (profilesRate != nil && tracesRate != nil) {
+            config.profilesSamplerDecision =
+                [[SentrySamplerDecision alloc] initWithDecision:kSentrySampleDecisionYes
+                                                  forSampleRate:profilesRate];
+            context.sampleRate = tracesRate;
+        }
+        launchTracer = [[SentryTracer alloc] initWithTransactionContext:context
+                                                                    hub:nil
+                                                          configuration:config];
```

Comment:
`m`: I wonder if it makes sense to start the tracer when the `profilesRate` is `nil` or the `tracesRate` is `nil`. Currently, we would start the tracer.

Reply:
Ah! I refactored this area to help write tests for this one https://github.com/getsentry/sentry-cocoa/pull/3636#discussion_r1489050618

Check out where I early return there now: https://github.com/getsentry/sentry-cocoa/blob/2fdf6e96851e948183a7f0998aec546e68a2ee74/Sources/Sentry/Profiling/SentryLaunchProfiling.m#L137-L149